File: mine/src/main/java/com/leory/badminton/mine/mvp/ui/activity/LoginActivity.java
package com.leory.badminton.mine.mvp.ui.activity;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.text.TextUtils;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;

import com.leory.badminton.mine.R;
import com.leory.badminton.mine.R2;
import com.leory.badminton.mine.di.component.DaggerLoginComponent;
import com.leory.badminton.mine.mvp.contract.LoginContract;
import com.leory.badminton.mine.mvp.model.bean.UserInfoBean;
import com.leory.badminton.mine.mvp.model.sp.AccountSp;
import com.leory.badminton.mine.mvp.presenter.LoginPresenter;
import com.leory.commonlib.base.BaseActivity;
import com.leory.commonlib.base.delegate.IComponent;
import com.leory.commonlib.di.component.AppComponent;
import com.leory.commonlib.di.scope.ActivityScope;
import com.leory.commonlib.utils.LogUtils;
import com.leory.commonlib.utils.ToastUtils;
import com.leory.commonlib.widget.XSDToolbar;
import com.umeng.socialize.UMAuthListener;
import com.umeng.socialize.UMShareAPI;
import com.umeng.socialize.bean.SHARE_MEDIA;

import java.util.Map;

import butterknife.BindView;
import butterknife.OnClick;

/**
 * Describe : 登陆activity
 * Author : leory
 * Date : 2019-06-19
 */
@ActivityScope
public class LoginActivity extends BaseActivity<LoginPresenter> implements LoginContract.View {
    @BindView(R2.id.toolbar)
    XSDToolbar toolbar;
    @BindView(R2.id.et_phone)
    EditText etPhone;
    @BindView(R2.id.et_code)
    EditText etCode;
    @BindView(R2.id.txt_code)
    TextView txtCode;
    @BindView(R2.id.et_pwd)
    EditText etPwd;

    public static void launch(Activity preActivity) {
        preActivity.startActivity(new Intent(preActivity, LoginActivity.class));
    }

    @Override
    public IComponent setupActivityComponent(IComponent component) {
        DaggerLoginComponent.builder().appComponent((AppComponent) component)
                .view(this)
                .build()
                .inject(this);
        return null;
    }

    @Override
    public int initView(@Nullable Bundle savedInstanceState) {
        return R.layout.activity_login;
    }

    @Override
    public void initData(@Nullable Bundle savedInstanceState) {
        toolbar.setOnBackListener(new XSDToolbar.OnBackListener() {
            @Override
            public void onBackClick() {
                finish();
            }
        });
    }

    @OnClick({R2.id.wechat, R2.id.qq, R2.id.btn_login})
    public void onViewClicked(View view) {
        if (view.getId() == R.id.wechat) {
            getThirdPartiesInfo(SHARE_MEDIA.WEIXIN);
        } else if (view.getId() == R.id.qq) {
            getThirdPartiesInfo(SHARE_MEDIA.QQ);
        } else if (view.getId() == R.id.btn_login) {
            String phone = etPhone.getText().toString().trim();
            String pwd = etPwd.getText().toString().trim();
            if (TextUtils.isEmpty(phone)) {
                showMessage("请输入正常的手机号码");
            } else if (TextUtils.isEmpty(pwd)) {
                showMessage("请输入密码");
            } else {
                presenter.login(phone, pwd);
            }

        }
    }

    private void getThirdPartiesInfo(SHARE_MEDIA share_media) {
        UMShareAPI.get(this).getPlatformInfo(this, share_media, new UMAuthListener() {
            @Override
            public void onStart(SHARE_MEDIA share_media) {
                LogUtils.d(TAG, share_media.toString());
            }

            @Override
            public void onComplete(SHARE_MEDIA share_media, int i, Map<String, String> map) {
                LogUtils.d(TAG, map.toString());
                UserInfoBean bean = AccountSp.getUserInfoBean();
                if (bean == null) {
                    bean = new UserInfoBean();
                    bean.setWechatUid(map.get("uid"));
                    bean.setName(map.get("name"));
                    bean.setGender(map.get("gender"));
                    bean.setIconUrl(map.get("iconurl"));
                }
                if (share_media == SHARE_MEDIA.QQ) {
                    bean.setQqUid(map.get("uid"));
                } else if (share_media == SHARE_MEDIA.WEIXIN) {
                    bean.setWechatUid(map.get("uid"));
                }
                bean.setShareMedia(share_media);
                AccountSp.putUserInfoBean(bean);
                AccountSp.putLoginState(true);
                showMessage("登陆成功");
                finish();
            }

            @Override
            public void onError(SHARE_MEDIA share_media, int i, Throwable throwable) {
                LogUtils.d(TAG, throwable.toString());
            }

            @Override
            public void onCancel(SHARE_MEDIA share_media, int i) {
                LogUtils.d(TAG, "取消了");
            }
        });
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        UMShareAPI.get(this).onActivityResult(requestCode, resultCode, data);
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        UMShareAPI.get(this).release();

    }

    @Override
    protected void onSaveInstanceState(Bundle outState) {
        super.onSaveInstanceState(outState);
        UMShareAPI.get(this).onSaveInstanceState(outState);
    }

    @Override
    public void showMessage(@NonNull String message) {
        ToastUtils.showShort(message);
    }

    @Override
    public void loginSuccess(UserInfoBean bean) {
        AccountSp.putUserInfoBean(bean);
        AccountSp.putLoginState(true);
        showMessage("登陆成功");
        finish();
    }

}

File: mine/src/main/java/com/leory/badminton/mine/mvp/ui/activity/LoginActivity.kt
package com.leory.badminton.mine.mvp.ui.activity

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.text.TextUtils
import android.view.View
import com.leory.badminton.mine.R
import com.leory.badminton.mine.di.component.DaggerLoginComponent
import com.leory.badminton.mine.mvp.contract.LoginContract
import com.leory.badminton.mine.mvp.model.bean.UserInfoBean
import com.leory.badminton.mine.mvp.model.sp.AccountSp
import com.leory.badminton.mine.mvp.presenter.LoginPresenter
import com.leory.commonlib.base.BaseActivity
import com.leory.commonlib.base.delegate.IComponent
import com.leory.commonlib.di.component.AppComponent
import com.leory.commonlib.di.scope.ActivityScope
import com.leory.commonlib.utils.LogUtils
import com.leory.commonlib.utils.ToastUtils
import com.umeng.socialize.UMAuthListener
import com.umeng.socialize.UMShareAPI
import com.umeng.socialize.bean.SHARE_MEDIA
import kotlinx.android.synthetic.main.activity_login.*

/**
 * Describe : 登陆activity
 * Author : leory
 * Date : 2019-06-19
 */
@ActivityScope
class LoginActivity : BaseActivity<LoginPresenter>(), LoginContract.View {

    override fun setupActivityComponent(component: IComponent): IComponent? {
        DaggerLoginComponent.builder().appComponent(component as AppComponent)
                .view(this)
                .build()
                .inject(this)
        return null
    }

    override fun initView(savedInstanceState: Bundle?): Int = R.layout.activity_login


    override fun initData(savedInstanceState: Bundle?) {
        toolbar.setOnBackListener { finish() }
        wechat.setOnClickListener { onViewClicked(it) }
        qq.setOnClickListener { onViewClicked(it) }
        btn_login.setOnClickListener { onViewClicked(it) }
    }

    override fun showMessage(message: String) {
        ToastUtils.showShort(message)
    }

    override fun loginSuccess(bean: UserInfoBean) {
        AccountSp.userInfoBean = bean
        AccountSp.loginState = true
        showMessage("登陆成功")
        finish()
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        UMShareAPI.get(this).onActivityResult(requestCode, resultCode, data)
    }

    override fun onDestroy() {
        super.onDestroy()
        UMShareAPI.get(this).release()

    }

    override fun onSaveInstanceState(outState: Bundle) {
        super.onSaveInstanceState(outState)
        UMShareAPI.get(this).onSaveInstanceState(outState)
    }

    private fun onViewClicked(view: View) {
        when {
            view.id == R.id.wechat -> getThirdPartiesInfo(SHARE_MEDIA.WEIXIN)
            view.id == R.id.qq -> getThirdPartiesInfo(SHARE_MEDIA.QQ)
            view.id == R.id.btn_login -> {
                val phone = et_phone.text.toString().trim { it <= ' ' }
                val pwd = et_pwd.text.toString().trim { it <= ' ' }
                when {
                    TextUtils.isEmpty(phone) -> showMessage("请输入正常的手机号码")
                    TextUtils.isEmpty(pwd) -> showMessage("请输入密码")
                    else -> presenter!!.login(phone, pwd)
                }

            }
        }
    }

    private fun getThirdPartiesInfo(share_media: SHARE_MEDIA) {
        UMShareAPI.get(this).getPlatformInfo(this, share_media, object : UMAuthListener {
            override fun onStart(share_media: SHARE_MEDIA) {
                LogUtils.d(TAG, share_media.toString())
            }

            override fun onComplete(share_media: SHARE_MEDIA, i: Int, map: Map<String, String>) {
                LogUtils.d(TAG, map.toString())
                var bean = AccountSp.userInfoBean
                if (bean == null) {
                    bean = UserInfoBean()
                    bean.wechatUid = map["uid"]
                    bean.name = map["name"]
                    bean.gender = map["gender"]
                    bean.iconUrl = map["iconurl"]
                }
                if (share_media == SHARE_MEDIA.QQ) {
                    bean.qqUid = map["uid"]
                } else if (share_media == SHARE_MEDIA.WEIXIN) {
                    bean.wechatUid = map["uid"]
                }
                bean.shareMedia = share_media
                AccountSp.userInfoBean = bean
                AccountSp.loginState = true
                showMessage("登陆成功")
                finish()
            }

            override fun onError(share_media: SHARE_MEDIA, i: Int, throwable: Throwable) {
                LogUtils.d(TAG, throwable.toString())
            }

            override fun onCancel(share_media: SHARE_MEDIA, i: Int) {
                LogUtils.d(TAG, "取消了")
            }
        })
    }


    companion object {

        @JvmStatic
        fun launch(preActivity: Activity) {
            preActivity.startActivity(Intent(preActivity, LoginActivity::class.java))
        }
    }

}
